src/UserALEWebExtension/background.js

```diff
@@ -63,27 +65,44 @@ function filterUrl(log) {
   return false;
 }
 
+// Sets http session id of tab logs to that of the target tab
+function injectSessions(log) {
+    let id = log.details.id;
+    if(id in tabToHttpSession) {
+      log.httpSession = tabToHttpSession[id];
+    } else {
+      log.httpSession = null
+    }
+    log.browserSessionId = browserSessionId;
+    return log;
+}
+
 browser.storage.local.get(defaultConfig, (res) => {
   // Apply url filter to logs generated by the background page.
-  userale.addCallbacks({filterUrl});
+  userale.addCallbacks({filterUrl, injectSessions});
   updateConfig(res);
+  browserSessionId = JSON.parse(window.sessionStorage.getItem('userAleHttpSessionId'));
```

Comment:
Is this the only place where the browserSessionId is set?

Why does this come from userAleHTTPSessionId?

Reply:
Yes it is the only place, and it comes from userAleHTTPSessionId because browser session is equivalent to the http session of the background page.